modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/ValueProvidersParameterDeclarationEnricher.java

```diff
@@ -248,6 +250,7 @@ private void enrichParameterFields(List<FieldValues> fieldsValues, ParameterDecl
           .ifPresent(field -> requiresConfiguration.set(true));
 
       int partOrder = 1;
+      String providerName = name + "_" + hash(fieldValues.targetSelectors());
```

Comment:
I know that it is a long shot but, what happens if there is a collision? 
Also, I would change the suffix to #hash or $hash, or directly <hash>, I think that it looks better. And, if possible, only use positive numbers.
Just a comment, not required for approval.

Reply:
I would externalize the logic of generating and handling providerNames with suffix so that it's together.